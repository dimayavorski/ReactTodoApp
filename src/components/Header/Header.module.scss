.header {
	display: flex;
	align-items: center;
	justify-content: space-between;

	.title {
		font-size: 30px;
	}

	.themeToggle {
		display: flex;
		gap: 15px;
	}
}

