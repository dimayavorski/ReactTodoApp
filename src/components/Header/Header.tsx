import { Switch } from 'antd';
import { ThemeContext, themes } from '../../hooks/ThemeContext';
import { useState } from 'react';
import { MdDarkMode, MdLightMode } from 'react-icons/md';
import styles from './Header.module.scss';

export function Header() {
	const [darkMode, setDarkMode] = useState(true);
	return (
		<div className={styles.header}>
			<h3 className={styles.title}>TODO App</h3>
			<div className={styles.themeToggle}>
				<ThemeContext.Consumer>
					{({ changeTheme }) => (
						<Switch
							checked={darkMode}
							onChange={() => {
								setDarkMode((prev) => !prev);
								changeTheme(darkMode ? themes.light : themes.dark);
							}}
						/>
					)}
				</ThemeContext.Consumer>
				{darkMode ? <MdDarkMode /> : <MdLightMode />}
			</div>
		</div>
	);
}
