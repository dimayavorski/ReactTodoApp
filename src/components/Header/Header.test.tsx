import { Header } from './Header';
import { RenderResult, render } from '@testing-library/react';

describe('header tests', () => {
	const component: RenderResult = render(<Header />);

	it('should match snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});
