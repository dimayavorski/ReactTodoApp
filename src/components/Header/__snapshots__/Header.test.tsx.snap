// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <h3
          class="title"
        >
          TODO App
        </h3>
        <div
          class="themeToggle"
        >
          <button
            aria-checked="true"
            class="ant-switch css-dev-only-do-not-override-15rg2km ant-switch-checked"
            role="switch"
            type="button"
          >
            <div
              class="ant-switch-handle"
            />
            <span
              class="ant-switch-inner"
            >
              <span
                class="ant-switch-inner-checked"
              />
              <span
                class="ant-switch-inner-unchecked"
              />
            </span>
          </button>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 3a9 9 0 109 9c0-.46-.04-.92-.1-1.36a5.389 5.389 0 01-4.4 2.26 5.403 5.403 0 01-3.14-9.8c-.44-.06-.9-.1-1.36-.1z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <h3
        class="title"
      >
        TODO App
      </h3>
      <div
        class="themeToggle"
      >
        <button
          aria-checked="true"
          class="ant-switch css-dev-only-do-not-override-15rg2km ant-switch-checked"
          role="switch"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            <span
              class="ant-switch-inner-checked"
            />
            <span
              class="ant-switch-inner-unchecked"
            />
          </span>
        </button>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 3a9 9 0 109 9c0-.46-.04-.92-.1-1.36a5.389 5.389 0 01-4.4 2.26 5.403 5.403 0 01-3.14-9.8c-.44-.06-.9-.1-1.36-.1z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
